_posts/2019-12-16-database-migration-in-self-contained-systems.md

```diff
@@ -9,62 +9,67 @@ image:
+a change to its configuration?
+
+This article shows a way of implementing continuous delivery for database configuration using Kubernetes and Flyway. 
+
+{% include github-project.html url="https://github.com/arkuksin/scs-db-versioning" %}
+
+## The Problem
+When developing a self-contained system (SCS), we may want to have the whole database inside the SCS to avoid tight coupling
+to other systems. If we want to deliver an SCS, then our goal is to start the SCS with one click. To achieve this goal, we have to:
+
+ 1. set up an empty database,
+ 2. configure the database for access by the application,
+ 3. deploy the application and connect to the database from the application.
+
+We'll want to automate all those steps to create a smooth delivery pipeline to the customer. Technologies like docker make it easy 
+to set up an empty database for step 1. Also, we can easily deploy an application as a docker container for step 3. But how can we automatically
+configure the database and how can we control different versions for this configuration?
+
+Using the declarative approach of Kubernetes as a container orchestration system, we can easily build up the automated process to deliver the SCS.
+We can just declare the desired state of components like a database or application and Kubernetes takes care of the rest.
+But it's not possible to declare the desired database configuration with its schemas, users, privileges and so on, because it is the *internal* configuration
 of the database.
 
 Imagine we have a pipeline to deliver our SCS. This pipeline does the three steps from the list above.
-If we start the pipeline first time, the database will be created, the database will be configured and the application will be deployed.
-If we later make a change in the business logic, the pipeline will run these steps und Kubernetes will not detect any changes on the database, but on the business logic application.
+When we start the pipeline for the first time, the database will be created and configured and the application will be deployed.
+If we later make a change in the business logic, the pipeline will run these steps and Kubernetes will not detect any changes on the database, but on the business logic application.
 In this case, Kubernetes is responsible for updating our application. It works fine with Kubernetes. 
 
-But we want also to be able to make changes on one script for the database configuration. After that we want, that this change is detected and
-the database configuration will be updated.
-
-In this article, we will have a look at how to configure the database in a continuous way and how to have control over the version of the database configuration.
-It means, we want to be able to update the database configuration in the continuous delivery process. 
+**But we also want to be able to make changes to the database configuration (e.g. to change some permissions) and have that change deploy automatically**.
```

Comment:
> (e.g. to change some permissions)

this is fine, but I just want to show in this comment another example from my experience. I added a quartz framework to my application and I wanted to add a new scheme for that to split the business data and quartz data. This was a good use case to apply the approach from this article.